### PR TITLE
Add tests to r2 score metric

### PR DESCRIPTION
**Pull request by oneraghavan**
Adding test for r2 score metric

### CODE REVIEW

File: tests/test_metrics.py [hunk 1]
```diff
@@ -175,6 +175,18 @@ def test_explained_variance(p, t, expect):
...
+@pytest.mark.parametrize("p, t, expect", [
+    (torch.arange(-5, 5).float(), torch.arange(-5,  5).float(), 1),
+    (torch.zeros(10).float(), torch.arange(-5, 5).float(), -0.0303),
+    (torch.arange(-5, 5).float(), torch.zeros(10).float(), -float("inf")),
+    (p1/2., p1, 0.6875),
+    (p1, t2, -2.75),
+])
+def test_r2_score(p, t, expect):
+    this_tests(r2_score)
+    print(r2_score(p, t), expect)
```
Comment (sgugger) [inline tests/test_metrics.py]: No print statements in the tests please. I can understand you needed it for debugging but let's remove it for the final version ;)

Comment (oneraghavan) [reply tests/test_metrics.py]: Sure , Done.

**Comment by sgugger** [comment]
Thanks!
